### PR TITLE
xtest: regression: add more test cases for SHA-3 inside 4001

### PR DESCRIPTION
**Pull request by ldshyu**
Long test cases for SHA3-224/256/384/512 and SHAKE128/256 are included

### CODE REVIEW

File: host/xtest/regression_4000.c [hunk 1]
```diff
@@ -1009,6 +1009,172 @@ static const uint8_t hash_data_shake256_out1[] = {
...
+	0xa5, 0x2f, 0xbb, 0xe7, 0x11, 0x85, 0x4a, 0x44,
+	0x05, 0x54, 0x29, 0x5f, 0xfb, 0x7a, 0x79, 0x69,
+	0xa1, 0x79, 0x08, 0xd1, 0x01, 0x63, 0xbf, 0xbe,
+	0x8f, 0x1d, 0x52, 0xa6, 0x76, 0xe8, 0xa0, 0x13,
+	0x7b, 0x56, 0xa1, 0x1c, 0xdf, 0x0f, 0xfb, 0xb4,
+	0x56, 0xbc, 0x89, 0x9f, 0xc7, 0x27, 0xd1, 0x4b,
+	0xd8, 0x88, 0x22, 0x32, 0x54, 0x9d, 0x91, 0x4e,
+};
+
+static const uint8_t hash_data_shake128_in2[] = {
```
Comment (jenswi-linaro) [inline host/xtest/regression_4000.c:1120]: Where did you find this test vector?

File: host/xtest/regression_4000.c [hunk 2]
```diff
@@ -1009,6 +1009,172 @@ static const uint8_t hash_data_shake256_out1[] = {
 	0x9f, 0x87, 0x26, 0xe4, 0x62, 0xa1
 };
 
+/* more SHA3 test cases from
```
Comment (jenswi-linaro) [inline host/xtest/regression_4000.c:1012]:
``` C
/*
 * Please format multiline
 * comments like this.
 */

File: host/xtest/regression_4000.c (hunk 1 above)
Comment (ldshyu) [reply host/xtest/regression_4000.c:1121]: Nist website:
https://csrc.nist.gov/projects/cryptographic-algorithm-validation-program/secure-hashing

Comment (ldshyu) [reply host/xtest/regression_4000.c:1133]: Done.  Comments updated with where the test vectors came from.  Thanks.

**Comment by etienne-lms** [comment]
@jforissier, could you enable CI on this P-R?

File: host/xtest/regression_4000.c [hunk 3]
```diff
@@ -1009,6 +1009,186 @@ static const uint8_t hash_data_shake256_out1[] = {
...
 
+/*
+ * More SHA-3 test cases from
+ * https://csrc.nist.gov/projects/cryptographic-algorithm-validation-program/secure-hashing
+ * https://csrc.nist.gov/CSRC/media/Projects/Cryptographic-Algorithm-Validation-Program/documents/sha3/sha-3bytetestvectors.zip
+ */
+
+/* Excerpt from SHA3_224LongMsg.rsp */
+static const uint8_t hash_data_sha3_224_in2[] = {
+	0x31, 0xc8, 0x2d, 0x71, 0x78, 0x5b, 0x7c, 0xa6, 0xb6, 0x51, 0xcb, 0x6c, 0x8c, 0x9a, 0xd5, 0xe2,
```
Comment (jenswi-linaro) [inline host/xtest/regression_4000.c:1020]: These lines are too long, please keep them under 80, with an exception for the links of course.

**Comment by ldshyu** [comment]
commit message updated

**Pull request merged**